splunk_add_on_ucc_framework/install_python_libraries.py

```diff
@@ -58,30 +71,83 @@ def _subprocess_call(
     try:
-        return_code = _subprocess_call(command=cmd, command_desc=command_desc)
+        return_code = _subprocess_run(command=cmd, command_desc=command_desc).returncode
         if return_code != 0:
             raise CouldNotInstallRequirements
     except OSError as e:
         raise CouldNotInstallRequirements from e
 
 
 def _pip_is_lib_installed(
-    installer: str, target: str, libname: str, version: Optional[str] = None
+    installer: str,
+    target: str,
+    libname: str,
+    version: Optional[str] = None,
+    allow_higher_version: bool = False,
 ) -> bool:
+    if not version and allow_higher_version:
+        raise InvalidArguments(
+            "Parameter 'allow_higher_version' can not be set to True if 'version' parameter is not provided"
+        )
+
     lib_installed_cmd = f"{installer} -m pip show --version {libname}"
-    lib_version_match_cmd = f'{lib_installed_cmd} | grep "Version: {version}"'
 
-    cmd = lib_version_match_cmd if version else lib_installed_cmd
+    if version and allow_higher_version:
+        cmd = f'{lib_installed_cmd} | grep "Version"'
+    elif version and not allow_higher_version:
+        cmd = f'{lib_installed_cmd} | grep "Version: {version}"'
+    else:
+        cmd = lib_installed_cmd
 
     try:
         my_env = os.environ.copy()
         my_env["PYTHONPATH"] = target
-        return_code = _subprocess_call(command=cmd, env=my_env)
-        return return_code == 0
+        if allow_higher_version:
+            result = _subprocess_run(command=cmd, env=my_env)
+            if result.returncode != 0:
+                return False
+            result_version = result.stdout.decode("utf-8").split("Version:")[1].strip()
+            return Version(result_version) >= Version(version)
+        else:
+            return_code = _subprocess_run(command=cmd, env=my_env).returncode
+            return return_code == 0
     except OSError as e:
         raise CouldNotInstallRequirements from e
 
 
+def _check_ucc_library_in_requirements_file(path_to_requirements: str) -> bool:
+    with open(path_to_requirements) as f_reqs:
+        content = f_reqs.readlines()
+    for line in content:
+        if "splunktaucclib" in line:
+            return True
+    return False
+
+
+def _check_libraries_required_for_ui(
+    python_binary_name: str, ucc_lib_target: str, path_to_requirements_file: str
+) -> None:
+    if not _pip_is_lib_installed(
+        installer=python_binary_name,
+        target=ucc_lib_target,
+        libname="splunktaucclib",
+    ):
+        raise SplunktaucclibNotFound(
+            f"This add-on has an UI, so the splunktaucclib is required but not found in "
+            f"{path_to_requirements_file}. Please add it there and make sure it is at least version 6.4."
+        )
+    if not _pip_is_lib_installed(
+        installer=python_binary_name,
+        target=ucc_lib_target,
+        libname="splunktaucclib",
+        version="6.4",
```

Comment:
Should we keep it as a constant somewhere on top of the file? Also, should it be `6.4.0`?

Reply:
Right, I will move it to the top. Yes, it's version [6.4.0 ](https://github.com/splunk/addonfactory-ucc-library/releases/tag/v6.4.0)